src/Symfony/Component/Messenger/Worker.php

```diff
@@ -254,15 +254,17 @@ private function flush(bool $force): bool
 
         $this->unacks = new \SplObjectStorage();
 
-        foreach ($unacks as $batchHandler) {
+        while ($unacks->valid()) {
+            $batchHandler = $unacks->current();
             [$envelope, $transportName] = $unacks[$batchHandler];
             try {
                 $this->bus->dispatch($envelope->with(new FlushBatchHandlersStamp($force)));
                 $envelope = $envelope->withoutAll(NoAutoAckStamp::class);
-                unset($unacks[$batchHandler], $batchHandler);
             } catch (\Throwable $e) {
                 $this->acks[] = [$transportName, $envelope, $e];
             }
+            $unacks->next();
+            $unacks->detach($batchHandler);
```

Comment:
This means that we now don't call the destructor of the $batchHandler (if any) inside the try/catch.
Is that the fix or a regression?

Reply:
The fix is that we unset the batchHandler after we set the iterator to the next batchHandler. Before it would unset the current batchHandler, and then tries to proceed to the next batchHandler but can't find the current index because it was unset.

See this comment: https://www.php.net/manual/en/splobjectstorage.detach.php#97644

Catching a Throwable of the unset, has nothing to do with the handling of the batch.

Reply:
@nicolas-grekas Do you have enough information to accept this PR or do you need more information. It would help us a lot if this bug is fixed.